src/Illuminate/Foundation/Support/Providers/EventServiceProvider.php

```diff
@@ -48,4 +51,58 @@ public function listens()
     {
         return $this->listen;
     }
+
+    /**
+     * Get the discovered events and listeners for the application.
+     *
+     * @return array
+     */
+    protected function discoveredEvents()
+    {
+        if ($this->app->eventsAreCached()) {
+            return require $this->app->getCachedEventsPath();
+        }
+
+        return $this->shouldDiscoverEvents()
```

Comment:
`shouldDiscoverEvents()` is redundant because `discoverEventsWithin()` can return empty array. The non-empty array `$this->app->path('Listeners')` can be returned from `App\Providers\EventServiceProvider` and the feature can be enabled for new Laravel installations by default.